### PR TITLE
Task queues

### PR DESCRIPTION
**Pull request by flossypurse**
## What does this PR do?

Per docs plan:
- Refreshed basic Task Queue concept page: https://deploy-preview-427--mystifying-fermi-1bc096.netlify.app/docs/concepts/task-queues

- Explicitly adds implementation information for Task Queues in SDK docs for:
    - Go: https://deploy-preview-427--mystifying-fermi-1bc096.netlify.app/docs/go/task-queues
    - Java: https://deploy-preview-427--mystifying-fermi-1bc096.netlify.app/docs/java/task-queues
    - PHP: https://deploy-preview-427--mystifying-fermi-1bc096.netlify.app/docs/php/task-queues

Shared "high level" Task Queue description  is used.

### CODE REVIEW

**Review by wolfy-j — APPROVED** [review]

**Review by tsurdilo — APPROVED** [review]

File: docs/concepts/task-queues.md [hunk 1]
```diff
@@ -4,25 +4,70 @@ title: Task Queues
...
+Temporal Task Queues are a little bit different from commonly used queuing technologies.
+The main difference being that they do not require explicit registration and are created on demand.
+Task Queues are very lightweight and there is no limit to the total number of Task Queues that the system can handle.
+
+There are multiple advantages of using a Task Queue to deliver Tasks to a Worker, instead of invoking actions via a synchronous RPC.
+
+- Workers do not need to have any open ports, which is more secure.
+- Workers do not need to advertise themselves through DNS or any other network discovery mechanism.
+- When all Workers are down, messages simply persist in a Task Queue, waiting for the Workers to recover.
 - A Worker polls for a message only when it has spare capacity, so it never gets overloaded.
```
Comment (Sushisource) [inline docs/concepts/task-queues.md:40]: I realize this is the one line that didn't change, lol, but "never" might be too strong of working here. Users could still do weird things in their code causing workers to slow down, even with only one task, for example.

```suggestion
- A Worker polls for a message only when it has spare capacity, avoiding overloading itself.
```

File: docs/concepts/task-queues.md [hunk 2]
```diff
@@ -4,25 +4,70 @@ title: Task Queues
...
+:::note
+
+All Workers listening to a given Task Queue must have identical registration of Activities/Workflows.
+The one exception to this is during a Server upgrade, where it is okay to have registration temporarily misaligned while the binary rolls out.
+
+:::
+
+## Task routing
+
+File processing tends to be one of the best examples of why you might need to route a Task to a specific process.
```
Comment (Sushisource) [inline docs/concepts/task-queues.md:54]: The Go example, at least, uses sessions to accomplish this. Java and PHP do indeed appear to do some task queue naming stuff to make it work, but at minimum having Go in here is slightly confusing b/c it manages it a different (more abstracted) way.

We probably at least want to call that out, or possibly use a different example

**Comment by Sushisource** [comment]
Also just heads up @flossypurse I also recently documented this at a much lower level here: https://github.com/temporalio/sdk-core/blob/master/arch_docs/sticky_queues.md and I want to do this generally for all of the complex sdk<->server interactions long run. We should think about where those kinds of docs can fit into our overall picture long run.

File: docs/concepts/task-queues.md (hunk 1 above)
Comment (flossypurse) [reply docs/concepts/task-queues.md:40]: SG!

File: docs/concepts/task-queues.md (hunk 2 above)
Comment (flossypurse) [reply docs/concepts/task-queues.md:54]: Ya, good catch!

File: docs/concepts/task-queues.md [hunk 3]
```diff
@@ -4,25 +4,72 @@ title: Task Queues
...
 
-Temporal calls this queue used to dispatch Activity Tasks an Activity Task Queue.
+import SharedTaskQueuesBasic from '../shared/task-queues-basic.md'
 
-Similarly, when a Workflow needs to handle an external event, a Command Task is created.
-A Command Task Queue is used to deliver it to the Worker.
+<SharedTaskQueuesBasic
+workflowLink="/docs/concepts/workflows"
+workerLink="/docs/concepts/workers"
+/>
```
Comment (sw-yx) [inline docs/concepts/task-queues.md:17]: niiiice

**Pull request merged**